pyrene/src/components/DateTimePicker/DateTimeInput/DateTimeInput.tsx

```diff
@@ -0,0 +1,138 @@
+
+import React, { forwardRef } from 'react';
+import clsx from 'clsx';
+import { allowedValueCheck } from '../../../utils/DateUtils';
+
+import Icon from '../../Icon/Icon';
+import styles from './dateTimeInput.css';
+
+export interface DateTimeInputProps {
+  /**
+   * Date format used by component
+   */
+  dateFormat?: string,
+  /**
+   * Boolean to toggle time display
+  */
+  dateOnly?: boolean,
+  /**
+   * Input date value
+  */
+  dateValue?: string,
+  /**
+   * Input error
+  */
+  errorValue?: string,
+  /**
+   * Component is disabled
+  */
+  disabled?: boolean,
+  /**
+   * Input field label
+  */
+  label?: string,
+  /**
+   * Input field name
+  */
+  name?: string,
+  /**
+   * Onchange from parent. Currently this handles onchange function passed from react-datepicker
+  */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  /**
+   * Onclick from parent. Currently this handles onclick function passed from react-datepicker
+  */
+  onClick?: () => void,
+  /**
+   * Onfocus from parent.
+  */
+  onFocus?: () => void,
+  /**
+   * Callback to set Date value in parent
+   */
+  setDateValue?: (value: string) => void,
+  /**
+   * Callback to set Time value in parent
+   */
+  setTimeValue?: (value: string) => void,
+  /**
+   * Time format used by component
+   */
+  timeFormat?: string,
+  /**
+   * Input time value
+  */
+  timeValue?: string,
+}
+
+const DateTimeInput = forwardRef(({
+  dateFormat = 'dd.MM.yyyy',
+  dateOnly = false,
+  dateValue = '',
+  errorValue,
+  disabled,
+  label = dateOnly ? 'Date' : 'Date & Time',
+  name,
+  onChange,
+  onClick,
+  onFocus,
+  setDateValue,
+  setTimeValue,
+  timeFormat = ' HH:mm',
+  timeValue = '',
+}:DateTimeInputProps, ref:React.Ref<HTMLInputElement>) => {
+
+  const handleDateOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
The behavior of the input box feels a bit odd:
- inputting `abcd` won't show anything as it doesn't match the regex
- inputting `1234` would be considered valid input but it's apparently not a good time format
- `111111111` is valid but `1111111111` is not valid as this is 1 more digit

large part of this behavior comes from the regex and the logic about `handleDataChange` here, let's discuss about it later :)

Reply:
As discussed we will be keeping the current validation style.